src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+	Avatar,
+	Button,
+	Dropdown,
+	DropdownButton,
+	DropdownContent,
+	Icon,
+	IconName,
+	MenuContent,
+	TextInput,
+} from '@viaa/avo2-components';
+import { isEmpty, isNil, truncate } from 'lodash';
+import React, { FC, useState } from 'react';
+
+import withUser, { UserProps } from '../../hocs/withUser';
+import useTranslation from '../../hooks/useTranslation';
+
+import { shareUserRightToString, sortShareUsers } from './ShareWithColleagues.helpers';
+import './ShareWithColleagues.scss';
+import { ShareUserInfo, ShareUserInfoRights } from './ShareWithColleagues.types';
+
+type ShareWithColleaguesProps = {
+	users: ShareUserInfo[];
+	onAddNewUser: (info: Partial<ShareUserInfo>) => void;
+	onEditRights: (info: ShareUserInfo, newRights: ShareUserInfoRights) => void;
+	onDeleteUser: (info: ShareUserInfo) => void;
+};
+
+const ShareWithColleagues: FC<ShareWithColleaguesProps & UserProps> = ({
+	users,
+	user,
+	onAddNewUser,
+	onDeleteUser,
+	onEditRights,
+}) => {
+	const { tText } = useTranslation();
+	const currentUser = users.find((u) => u.email === user?.mail) as ShareUserInfo;
+	const [isRightsDropdownOpen, setIsRightsDropdownOpen] = useState<boolean>(false);
+	const [newShareUser, setNewShareUser] = useState<Partial<ShareUserInfo>>({
+		email: undefined,
+		rights: undefined,
+	});
+	const [error, setError] = useState<string | null>(null);
+
+	const handleRightsButtonClicked = () => {
+		setIsRightsDropdownOpen(!isRightsDropdownOpen);
+	};
+
+	const handleAddNewUser = () => {
+		if (/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]+$/.test(newShareUser.email as string)) {
+			onAddNewUser(newShareUser);
+			setError(null);
+		} else {
+			setError(
+				tText(
+					'shared/components/share-with-colleagues/share-with-colleagues___email-is-geen-geldig-emailadres'
+				)
+			);
+		}
+	};
+
+	const handleEditUserRights = (info: ShareUserInfo, newRights: ShareUserInfoRights) => {
+		onEditRights(info, newRights);
+	};
+
+	const handleDeleteUser = (info: ShareUserInfo) => {
+		onDeleteUser(info);
+	};
+
+	const updateNewShareUserInfo = (value: Record<string, string>) => {
+		setNewShareUser({
+			...newShareUser,
+			...value,
+		});
+	};
+
+	const renderColleaguesInfoList = () => {
+		if (users.length > 0) {
+			return (
+				<ul className="c-colleagues-info-list">
+					{sortShareUsers(users).map((user, index) => (
```

Comment:
the user variable name is a naming conflict with the user parameter of the component
this doesn't break anything, but it is better to avoid it since it can cause confusion
```suggestion
					{sortShareUsers(users).map((contributorUser, index) => (
```